java/meterpreter/stdapi/src/main/java/com/metasploit/meterpreter/ProcessChannel.java

```diff
@@ -26,6 +26,30 @@ public ProcessChannel(Meterpreter meterpreter, Process process) {
         new StderrThread(err).start();
     }
 
+    /**
+     * Read at least one byte, and up to maxLength bytes from this stream.
+     *
+     * @param maxLength The maximum number of bytes to read.
+     * @return The bytes read, or <code>null</code> if the end of the stream has been reached.
+     */
+    public synchronized byte[] read(int maxLength) {
+        if (closed)
+            return null;
+        if (active)
+            throw new IllegalStateException("Cannot read; currently interacting with this channel");
+        if (!waiting || (toRead != null && toRead.length == 0))
+            return new byte[0];
+        if (toRead == null)
+            return null;
+        byte[] result = new byte[Math.min(toRead.length, maxLength)];
+        System.arraycopy(toRead, 0, result, 0, result.length);
+        byte[] rest = new byte[toRead.length - result.length];
+        System.arraycopy(toRead, result.length, rest, 0, rest.length);
+        toRead = rest;
+        notifyAll();
+        return result;
```

Comment:
This part (lines 42-50) could possible be replaced by

```suggestion
        return super.read(maxLength);
```

Otherwise LGTM